packages/features/ee/organizations/components/CreateANewOrganizationForm.tsx

```diff
@@ -0,0 +1,318 @@
+import { signIn } from "next-auth/react";
+import { useRouter } from "next/router";
+import type { Dispatch, SetStateAction } from "react";
+import { useState } from "react";
+import useDigitInput from "react-digit-input";
+import { Controller, useForm } from "react-hook-form";
+
+import { subdomainSuffix } from "@calcom/features/ee/organizations/lib/orgDomains";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import slugify from "@calcom/lib/slugify";
+import { telemetryEventTypes, useTelemetry } from "@calcom/lib/telemetry";
+import { trpc } from "@calcom/trpc/react";
+import {
+  Button,
+  Form,
+  TextField,
+  Alert,
+  Dialog,
+  DialogClose,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  Label,
+  Input,
+} from "@calcom/ui";
+import { ArrowRight, Info } from "@calcom/ui/components/icon";
+
+function extractDomainFromEmail(email: string) {
+  let out = "";
+  try {
+    const match = email.match(/^(?:.*?:\/\/)?.*?(?<root>[\w\-]*(?:\.\w{2,}|\.\w{2,}\.\w{2}))(?:[\/?#:]|$)/);
```

Comment:
NIT: since this is an email isnt this always just the same as `email.split("@")[1]` 

Or am i not understanding this regex correctly?

Reply:
user@example.com
regex output: example.com
split output: example.com

https://www.example.co.uk/path/to/page
regex output: example.co.uk 
split output: example.co.uk/path/to/page (however emails are never in this format?)

Reply:
This was meaning to extract the correct domain from subdomain names as well, such as `leo@engineering.cal.com` to be `cal.com`